### PR TITLE
refactor(frontend): Simplify props of Hero

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

The props given to `Hero` are somehow a bit confused since derived from routing. They were ok before, since the `Hero` was used in all routes. With the latest changes, the scope of the `Hero` was reduced to only the main page (tokens) and the transaction page.

To clean-up a bit we refactor the Hero to be a bit more obvious. In another PR, we will probably merge `Hero` and `HeroContent`.

# Changes

- Use navigations utils directly in `HeroContent` and `+layout`
- Remove deprecated props from Hero.

### CODE REVIEW

File: src/frontend/src/tests/lib/utils/nav.utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import { AppPath, ROUTE_ID_GROUP_APP } from '$lib/constants/routes.constants';
```
Comment (peterpeterparker) [inline src/frontend/src/tests/lib/utils/nav.utils.spec.ts:1]: This shouldn't be part of another PR?

File: src/frontend/src/routes/(app)/+layout.svelte [hunk 2]
```diff
@@ -15,18 +15,16 @@
...
-				? 'transactions'
-				: 'tokens';
+	let tokensRoute: boolean;
+	$: tokensRoute = isRouteTokens($page);
+
+	let transactionsRoute: boolean;
+	$: transactionsRoute = isRouteTransactions($page);
+
+	let showHero: boolean;
+	$: showHero = tokensRoute || transactionsRoute;
```
Comment (peterpeterparker) [inline src/frontend/src/routes/(app)/+layout.svelte:27]: Out of curiosity, not a call to action: we already have E2E screenshots of every pages? That way we can assert that indeed the hero is displayed only when needed.

Comment (AntonioVentilii) [reply src/frontend/src/routes/(app)/+layout.svelte:27]: I agree! But we don't have for all yet, I was working on it. I have the first PR for the transactions page already in draft:

- #3345

File: src/frontend/src/tests/lib/utils/nav.utils.spec.ts (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/tests/lib/utils/nav.utils.spec.ts]: Ah yes sure:

- PR #3358

Comment (peterpeterparker) [reply src/frontend/src/tests/lib/utils/nav.utils.spec.ts]: Not just the test, you mean the utils as well?

Comment (AntonioVentilii) [reply src/frontend/src/tests/lib/utils/nav.utils.spec.ts]: correct, this PR is just for the `Hero` logic

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx
